### ISSUE TITLE
Waiting for ssh login on 127.0.0.1 with user root to sshd on port => 7222

### ISSUE THREAD

**Issue by iojeda** [jedi4ever/veewee#935]
I launch the following command
```bash
veewee vbox box arch64 --force
```
And every goes smooth but after the execution of ssh.sh it launches reboot.sh
and then i get the mesage of the issue i tried to connect ssh manually to VM but i cant

last dump of the log....

```bash
Executing command: ./ssh.sh
sh-4.3# pacman -S --noconfirm openssh
resolving dependencies...
looking for inter-conflicts...

Packages (4): dnssec-anchors-20130320-1  ldns-1.6.17-1  libedit-20140213_3.1-1
              openssh-6.6p1-1

Total Download Size:    1.18 MiB
Total Installed Size:   6.08 MiB

:: Proceed with installation? [Y/n] 
:: Retrieving packages ...
 libedit-20140213_3....    90.6 KiB   281K/s 00:00 [######################] 100%
 dnssec-anchors-2013...   980.0   B  0.00B/s 00:00 [######################] 100%
 ldns-1.6.17-1-x86_64     427.8 KiB   450K/s 00:01 [######################] 100%
 openssh-6.6p1-1-x86_64   693.0 KiB   476K/s 00:01 [######################] 100%
(4/4) checking keys in keyring                     [######################] 100%
(4/4) checking package integrity                   [######################] 100%
(4/4) loading package files                        [######################] 100%
(4/4) checking for file conflicts                  [######################] 100%
(4/4) checking available disk space                [######################] 100%
(1/4) installing libedit                           [######################] 100%
(2/4) installing dnssec-anchors                    [######################] 100%
(3/4) installing ldns                              [######################] 100%
Optional dependencies for ldns
    libpcap: ldns-dpa tool
(4/4) installing openssh                           [######################] 100%
Optional dependencies for openssh
    xorg-xauth: X11 forwarding
    x11-ssh-askpass: input passphrase in X
sh-4.3# 
sh-4.3# # Make sure SSH is allowed
sh-4.3# echo "sshd:ALL" > /etc/hosts.allow
sh-4.3# 
sh-4.3# # And everything else isn't
sh-4.3# echo "ALL:ALL" > /etc/hosts.deny
sh-4.3# 
sh-4.3# # Make sure sshd starts on boot
sh-4.3# systemctl enable sshd.service
ln -s '/usr/lib/systemd/system/sshd.service' '/etc/systemd/system/multi-user.target.wants/sshd.service'
sh-4.3# exit
Waiting for ssh login on 127.0.0.1 with user root to sshd on port => 7222 to work, timeout=10000 sec
.
Executing command: ./reboot.sh
Received a disconnect; moving on

Waiting for ssh login on 127.0.0.1 with user root to sshd on port => 7222 to work, timeout=10000 sec
...................................................

```
At this point im unable to login through ssh to the VM
Any help to solve the issue will be much apreciated thanks.
(Sorry for my bad English :) )

**Comment by mpapis** [comment]
closing as there was no feedback in three months, please let me know if you can provide more details and I will reopen

**Comment by chrisRidgers** [comment]
Am currently having a play attempting this, have run into the same issue.

Logged in using the open virtual box window and checked systemctl's services and the sshd daemon is running.

**Comment by chrisRidgers** [comment]
on a second attempt, i ran into the same issue and noticed that systemctl reported the sshd daemon as dead.

running systemctl restart sshd set the daemon to running, but the script still hangs waiting for login...

Ideas / useful information I can provide?

**Comment by mpapis** [comment]
@chrisRidgers could be network configuration problem, can you check `netstat -tpln` on the VM?

**Comment by ibizaman** [comment]
Got exactly the same problem, same output.

I could not run `netstat -tpln` though. It is not installed by default and running `pacman -S net-tools` says the machine is not connected to the internet.

I checked the VM's network configuration and the NAT network adapter is enabled with a redirection rule for host port 7222 to guest 22.

What's weird is that there are other "Waiting for ssh login on [...]" commands that have worked before that step.

**Comment by mpapis** [comment]
my guess is the network interface changed it's name from the installation and the configuration is on wrong name, we have seen similar thing in other systems, can you try debug it from this info?

**Comment by chrisRidgers** [comment]
just found this:

# Chroot into the new system
arch-chroot /mnt <<ENDCHROOT
# Previously dhcpcd.service was enabled. However, in my testing it repeatedly
# failed to connect to the network on reboot. Enable dhcpcd@.service has worked
# in my case. My guess is that this is due to the line
# After=sys-subsystem-net-devices-%i.device
# in the service file.
# Restarting dhcpcd.service after boot or using Network Manager instead of
# dhcpcd also works
# Maybe a related bug report?
# https://bugs.freedesktop.org/show_bug.cgi?id=59964
# Replace this with a better fix, when available.
systemctl enable dhcpcd\@eth0.service
# Disable systemd's Predictable Network Interface Names
ln -s /dev/null /etc/udev/rules.d/80-net-name-slot.rules

within

https://github.com/jedi4ever/veewee/blob/master/templates/archlinux-x86_64/base.sh

To me this describes the problems we're having.  I'm currently rebuilding the box to see how far I get with this info, but if anyone can read this and identify a definite solution I'd be happy to hear it :)

**Comment by mpapis** [comment]
the last two lines look like related to what I was thinking about:
> ```bash
> # Disable systemd's Predictable Network Interface Names
> ln -s /dev/null /etc/udev/rules.d/80-net-name-slot.rules
> ```

**Comment by chrisRidgers** [comment]
Right so this is looking like the cause of the problem.  As a test i logged into the VM and restarted dhcpcd.  Sure enough, the install process resumed.

ip link showed me an interface name enp0s3 (or the ilk of).  So is that ln supposed to override that behaviour and revert back to the eth0 naming scheme?  Because if so it hasn't done so.

**Comment by mpapis** [comment]
yes it looks like it did not happen, you need to investigate why it failed, it's either not executed or moved for execution later on after ssh?

**Comment by chrisRidgers** [comment]
It failed because of this:
http://www.freedesktop.org/wiki/Software/systemd/PredictableNetworkInterfaceNames/

You disable the assignment of fixed names, so that the unpredictable kernel names are used again. For this, simply mask udev's rule file for the default policy: ln -s /dev/null /etc/udev/rules.d/80-net-setup-link.rules (since v209: this file was called 80-net-name-slot.rules in release v197 through v208)

symlink linked to an outdated file/ filename.

So once I updated that the build went smoothly (bar some issues with chef, but that's something else completely).

Still, even with it working its not something I feel entirely comfortable with (disabling systemd's more current naming structure...)  During the boot process I noticed that even during the initial install the network interface was set set to the new naming scheme, if it's consistent within virtual box I'll see if setting the

systemctl enable dhcpcd\@eth0.service

line can be switched to reflect it, and then drop the rule change change completely.

Thoughts?

**Comment by ibizaman** [comment]
Well I suppose the name of the predictable name can be retrieve with something like:
```
ip addr show label 'en*' | head -1 | cut -d' ' -f2 | sed 's/://'
```
The only thing is I'm not sure if matching on 'en*' is okay.

**Comment by mpapis** [comment]
@chrisRidgers the new naming convention is desired in real servers, in our case it is reverse, we want the first interface to be always the same `eth0`, could you provide pull request that has both paths handled?

@ibizaman go with first @chrisRidgers solution, playing with different names is to complicated and unnecessary when we can just disable it.

**Comment by chrisRidgers** [comment]
I'm currently in work, I'll submit the relevant pull requests when I get home tonight,  so the agreement is the fix to the ln command is the most appropriate here?

**Comment by chrisRidgers** [comment]
(For knowledges sake) changing base.sh's systemctl service to

systemctl enable dhcpcd\@enp0s3.service

Worked for me.  The comments in base.sh reference a bug/ issue with the empty dhcpcd\@ service.

**Comment by ibizaman** [comment]
@mpapis I agree with your point of view, but if we can avoid disabling it, why not?
@chipchilders In fact I posted my answer just before going to work so I didn't have time to test with something like:
``` bash
systemctl enable dhcpcd\@$(ip addr show label 'en*' | head -1 | cut -d' ' -f2 | sed 's/://').service
```

About my comment about "is en* a good test for the label", it seems it is: http://cgit.freedesktop.org/systemd/systemd/tree/src/udev/udev-builtin-net_id.c#n29

**Comment by mpapis** [comment]
the thing with persistent network interface names it can be a lot of things, matching `en*` might be not enough, but anything you can get working and create a pull request I will merge

**Comment by ibizaman** [comment]
Okay, I'll do something tonight :)

**Comment by mpapis** [comment]
fixed with #1034
